### PR TITLE
Enforce DRY in InchiGen.py

### PR DESCRIPTION
**Pull request by zjp**
There was some repeated code in InchiGen.py that I saw could use some cleanup. I have tested to ensure the changes still work on a previously passing molecule.

### CODE REVIEW

File: InchiGen.py [hunk 1]
```diff
@@ -130,13 +133,13 @@ def RestoreNumsSDF(f, fold, AuxInfo):
...
-            if conatom == oldHcons[b][1]:
-                amap.append(oldHcons[b][0])
+    for newHcon in newHcons:
+        conatom = newHcon[1]
+        for oldHcon in oldHcons:
+            if conatom == oldHcon[1]:
+                amap.append(oldHcon[0])
                 #remove the number, so that it doesn't get added twice
-                oldHcons[b][1] = 0
+                oldHcons[1] = 0
```
Comment (KristapsE) [inline InchiGen.py:142]: Shouldn't this be
oldHcon[1] = 0

**Comment by KristapsE** [comment]
The changes look fine, InchiGen.py contains some of the oldest code in DP4-AI, so probably the least pythonic. Could you please check the line I've highlighted and run a few more tests on a couple of molecules from our dataset to check, that the diastereomers are still generated accurately?

Sorry for taking so long to getting to the other pull requests, things are a bit busy at the moment.

File: InchiGen.py (hunk 1 above)
Comment (zjp) [reply InchiGen.py]: It should! I've now made this change.

**Comment by zjp** [comment]
That's a good test, I'll run a few. Are the SDF files/SMILES available anywhere?

**Comment by KristapsE** [comment]
https://www.repository.cam.ac.uk/handle/1810/303161

Here is the full dataset for our DP4-AI paper. Each calculation folder contains Compound_.sdf, which is the base structure, and then Compount_1.sdf, Compount_2.sdf, Compount_3.sdf, etc, which are the full set of diastereomers. The DP4-AI, given the Compound_.sdf, should generate the numbered set of sdfs and then use those to start calculations. If all the computational files are already present (from MacroModel and Gaussian in our dataset), it will just run the analysis without trying to calculate anything.

**Comment by zjp** [comment]
I've discovered a bug in the branch in this PR -- it's definitely not ready to merge.

**Comment by KristapsE** [comment]
It's alright, thanks for letting me know! I'll merge it when you're happy with it.

**Pull request closed without merging**